config/main.py

```diff
@@ -1357,12 +1357,67 @@ def apply_patch(ctx, patch_file_path, format, dry_run, ignore_non_yang_tables, i
             patch_as_json = json.loads(text)
             patch = jsonpatch.JsonPatch(patch_as_json)
 
+        results = {}
         config_format = ConfigFormat[format.upper()]
-        GenericUpdater().apply_patch(patch, config_format, verbose, dry_run, ignore_non_yang_tables, ignore_path)
+        if multi_asic.is_multi_asic():
+            # Initialize a dictionary to hold changes categorized by ASIC
+            changes_by_asic = {}
 
+            # Function to extract ASIC identifier from the change path
+            def extract_asic_id(path):
```

Comment:
>extract_asic_id [](http://example.com/codeflow?start=16&length=15)

Use some mature python library like jsonpointer ?

Reply:
This is a simple extraction which just  return the json patch scope.

Reply:
Understand the logic is simple, still python library like jsonpointer is better than adhoc logic in extract_scope function. For example, does your code hanlde `/localhostABC` properly?

Reply:
Makes sense, pushed new change with individual test for extract_scope only.